lang_java/tree_sitter/grammar_to_tree_sitter_ast.py

```diff
@@ -0,0 +1,128 @@
+import json
+
+with open("java_grammar.json", 'r') as fs:
+    data = json.load(fs)
+
+codegen = ""
+
+for a, aval in data["rules"].items():
+    if aval["type"] == "CHOICE":
+        codegen += '''
+type {type} = '''.format(type=a)
+        for b in aval["members"]:
+            if b["type"] not in ["PREC_DYNAMIC", "PREC_LEFT", "SEQ", "STRING", "ALIAS"]:
+                codegen +='''
+  | {constructor} of {field}'''.format(constructor=b['name'].split("_")[0].capitalize(),field=b['name'])
+        codegen += "\n"
+    elif aval["type"] == "TOKEN": #for literals
+        pass
+    elif aval["type"] == "REPEAT": #only for program.modifiers
+        pass
+    elif aval["type"] == "STRING": #don't need
+        pass
+    elif aval["type"] == "PREC":
+        pass
+    elif aval["type"] == "PREC_RIGHT":
+        pass
+    elif aval["type"] == "PREC_LEFT":
+        pass
```

Comment:
Lets have a list of strings called exclude types that we skip so that we can eliminate CST specific nodes from the tree